rfc/2025-07-22-conditional-enable.md

```diff
@@ -0,0 +1,195 @@
+# Conditional field for enabling resources and modules
+
+## Current status
+
+Issue: https://github.com/opentofu/opentofu/issues/1306
+
+> [!NOTE]
+> Every time we refer to *resources* in this RFC, we're talking about [managed resources](https://opentofu.org/docs/language/resources/), [data sources](https://opentofu.org/docs/language/data-sources/) and [ephemeral resources](https://github.com/opentofu/opentofu/issues/2834).
+
+Right now, OpenTofu supports conditional enable/disable of resources by using a workaround with `count`.
+This approach brings a few problems, such as adding indexes to resources that would be single instances, making it harder to manage using these indexes.
+
+This RFC proposes a new way to do this in a cleaner and more semantic manner by using a new field on the `lifecycle` block called `enabled`:
+
+```
+resource "aws_instance" "example" {
+  # ...
+
+  lifecycle {
+    enabled = var.enable_server
+  }
+}
+```
+
+This is available not only for resources, but for modules too:
+
+```
+module "modcall" {
+  source = "./mod"
+
+  lifecycle {
+    enabled = var.env == "production"
+  }
+}
+```
+
+## Proposed Solution
+
+1. Raise errors if a resource is used while being disabled and the dependent resources are not disabled too;
+2. Support for conditional enabling on:
+  1. Resources - Add a new field on the `lifecycle` block called `enabled`;
+  2. Modules - Add a new `lifecycle` block for modules but only supporting the new `enabled` field.
+
+
+## Technical Approach
+
+### `lifecycle` support on Modules
+
+The lifecycle block is currently not supported on `modules`, but in our docs we mention to users that it's reserved for future usage:
+
+> OpenTofu does not use the lifecycle argument. However, the lifecycle block is reserved for future versions.
+
+https://opentofu.org/docs/language/modules/syntax/#meta-arguments
+
+It would be the right timing to start supporting it, but only for the `enabled` field.
+
+## Migration from existing resources
+
+### Migration from `count` to `enabled`
+
+As mentioned before, users were adding support for enabling/disabling resources by using `count = var.enabled ? 1 : 0` and this is the most common case where they want to use an `enabled` field.
```

Comment:
Where was this mentioned before? I know you mentioned `count` in the intro but not a specific example